### ISSUE TITLE
webface_align_train.list

### ISSUE THREAD

**Issue by chidaidai** [wujiyang/Face_Pytorch#58]
请问下在哪里获取webface_align_train.list这个文件，谢谢了

**Comment by ruabuliuqiu** [comment]
**First step：**extract images from *rec(webface)
_code as follows：_
import os
import random
import logging
import sys
import numbers
import math
import sklearn
import datetime
import numpy as np
import cv2
import mxnet as mx
from mxnet import ndarray as nd
from mxnet import io
from mxnet import recordio
def main():
    root_dir = './images'
    path_imgrec = 'faces_emore/train.rec'
    path_imgidx = path_imgrec[0:-4]+".idx"
    imgrec = recordio.MXIndexedRecordIO(path_imgidx, path_imgrec, 'r')
    s = imgrec.read_idx(0)
    header, _ = recordio.unpack(s)
    print(header)
    try:
        os.mkdir(root_dir)
    except:
        pass
    a,b = int(header.label[0]), int(header.label[1])
    for i in range(a, b):
        s = imgrec.read_idx(i)
        header, _ = recordio.unpack(s)
        print(header)
        try:
            os.mkdir(root_dir + "/MS1M-ArcFace-" + str(i))
        except:
            pass
        c,d = int(header.label[0]), int(header.label[1])
        for j in range(c, d):
            s = imgrec.read_idx(j)
            header, img = recordio.unpack(s)
            print(header)
            img = mx.image.imdecode(img)
            img = cv2.cvtColor(img.asnumpy(), cv2.COLOR_BGR2RGB)
            cv2.imwrite(root_dir + "/MS1M-ArcFace-" + str(i) + "/" + str(j)+".jpg", img)
if __name__ == "__main__":
    main()
**Second step：**generate webface_align_train.list  
_code as follows：_
import os

def dataset_list(dataset_path,dataset_list):
     label_list = os.listdir(dataset_path)
     f=open(dataset_list,'w')
     k=0
     for i in label_list:
         label_path=os.path.join(dataset_path,i)
         if os.listdir(label_path):
            image_list=os.listdir(label_path)
            for j in image_list:
                image_path=os.path.join(label_path, j)
                f.write(image_path+'  '+str(k)+'\n')
         k=k+1
     f.close()

if __name__=='__main__':
    dataset = '/home/Desktop/webfaces_img'
    list = '/home/Desktop/webfaces.list'
    dataset_list(dataset, list)

**Comment by ruabuliuqiu** [comment]
> 请问下在哪里获取webface_align_train.list这个文件，谢谢了
**First step：**extract images from *rec(webface)
code as follows：
`import os
import random
import logging
import sys
import numbers
import math
import sklearn
import datetime
import numpy as np
import cv2
import mxnet as mx
from mxnet import ndarray as nd
from mxnet import io
from mxnet import recordio
def main():
    root_dir = './images'
    path_imgrec = 'faces_emore/train.rec'
    path_imgidx = path_imgrec[0:-4]+".idx"
    imgrec = recordio.MXIndexedRecordIO(path_imgidx, path_imgrec, 'r')
    s = imgrec.read_idx(0)
    header, _ = recordio.unpack(s)
    print(header)
    try:
        os.mkdir(root_dir)
    except:
        pass
    a,b = int(header.label[0]), int(header.label[1])
    for i in range(a, b):
        s = imgrec.read_idx(i)
        header, _ = recordio.unpack(s)
        print(header)
        try:
            os.mkdir(root_dir + "/MS1M-ArcFace-" + str(i))
        except:
            pass
        c,d = int(header.label[0]), int(header.label[1])
        for j in range(c, d):
            s = imgrec.read_idx(j)
            header, img = recordio.unpack(s)
            print(header)
            img = mx.image.imdecode(img)
            img = cv2.cvtColor(img.asnumpy(), cv2.COLOR_BGR2RGB)
            cv2.imwrite(root_dir + "/MS1M-ArcFace-" + str(i) + "/" + str(j)+".jpg", img)
if __name__ == "__main__":
    main()`

**Comment by ruabuliuqiu** [comment]
[https://github.com/QuasarLight/Pytorch_Face_Recognition-/blob/master/Utils/Datasets_Utils/generate_dataset_list.py](url)
It works，but first you need extract images from webface(train.rec)

**Comment by why111222** [comment]
> [https://github.com/QuasarLight/Pytorch_Face_Recognition-/blob/master/Utils/Datasets_Utils/generate_dataset_list.py](url) 可以，但首先你需要从webface(train.rec)中提取图像

您提供的链接打不开呀

**Comment by why111222** [comment]
> 请问下在哪里获取webface_align_train.list这个文件，谢谢了

请问您找到这个文件了吗
